tmt/base.py

```diff
@@ -4550,12 +4550,10 @@ def runs(self, id_: tuple[str, ...], keep: Optional[int]) -> bool:
             last_run._workdir_load(last_run._workdir_path)
             assert last_run.workdir is not None  # narrow type
             return self._clean_workdir(last_run.workdir)
-        all_workdirs = list(tmt.utils.generate_runs(self.workdir_root, id_))
+        all_workdirs = list(tmt.utils.generate_runs(self.workdir_root, id_, all_=True))
         if keep is not None:
             # Sort by modify time of the workdirs and keep the newest workdirs
-            all_workdirs.sort(
-                key=lambda workdir: (workdir / 'run.yaml').stat().st_mtime, reverse=True
-            )
+            all_workdirs.sort(key=lambda workdir: workdir.stat().st_ctime, reverse=True)
```

Comment:
With the change from `st_mtime` to `st_ctime`, the command above is no longer true: "Sort by *modify* time..."